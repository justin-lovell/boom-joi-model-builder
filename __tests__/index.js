'use strict';

var ModelBuilder = require('../index');

test('a blank code will result in an empty result', function() {

  var map = ModelBuilder().createMap();

  expect(map).toMatchSnapshot();

});

test('a 401 is added', function() {

  var map = ModelBuilder().add(401).createMap();

  expect(map).toMatchSnapshot();

});

test('a 401 and 500 is added', function() {

  var map = ModelBuilder().add(401).add(500).createMap();

  expect(map).toMatchSnapshot();

});