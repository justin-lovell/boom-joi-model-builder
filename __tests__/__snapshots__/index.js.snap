// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a 401 and 500 is added 1`] = `
Object {
  "401":  {
    "_description": null,
    "_examples": Array [],
    "_flags": Object {},
    "_inner": Object {
      "children": Array [
        Object {
          "key": "statusCode",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {
              "allowOnly": true,
              "presence": "required",
            },
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                Infinity,
                -Infinity,
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [
              Object {
                "arg": undefined,
                "func": [Function],
                "name": "integer",
                "options": undefined,
              },
            ],
            "_type": "number",
            "_unit": null,
            "_valids": Set {
              "_set": Array [
                401,
              ],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "error",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {
              "presence": "required",
            },
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "message",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {},
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "data",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {},
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
      ],
      "dependencies": Array [],
      "patterns": Array [],
      "renames": Array [],
    },
    "_invalids": Set {
      "_set": Array [],
    },
    "_meta": Array [],
    "_notes": Array [],
    "_refs": Array [],
    "_settings": null,
    "_tags": Array [],
    "_tests": Array [],
    "_type": "object",
    "_unit": null,
    "_valids": Set {
      "_set": Array [],
    },
    "isJoi": true,
  },
  "500":  {
    "_description": null,
    "_examples": Array [],
    "_flags": Object {},
    "_inner": Object {
      "children": Array [
        Object {
          "key": "statusCode",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {
              "allowOnly": true,
              "presence": "required",
            },
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                Infinity,
                -Infinity,
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [
              Object {
                "arg": undefined,
                "func": [Function],
                "name": "integer",
                "options": undefined,
              },
            ],
            "_type": "number",
            "_unit": null,
            "_valids": Set {
              "_set": Array [
                500,
              ],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "error",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {
              "presence": "required",
            },
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "message",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {},
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "data",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {},
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
      ],
      "dependencies": Array [],
      "patterns": Array [],
      "renames": Array [],
    },
    "_invalids": Set {
      "_set": Array [],
    },
    "_meta": Array [],
    "_notes": Array [],
    "_refs": Array [],
    "_settings": null,
    "_tags": Array [],
    "_tests": Array [],
    "_type": "object",
    "_unit": null,
    "_valids": Set {
      "_set": Array [],
    },
    "isJoi": true,
  },
}
`;

exports[`a 401 is added 1`] = `
Object {
  "401":  {
    "_description": null,
    "_examples": Array [],
    "_flags": Object {},
    "_inner": Object {
      "children": Array [
        Object {
          "key": "statusCode",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {
              "allowOnly": true,
              "presence": "required",
            },
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                Infinity,
                -Infinity,
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [
              Object {
                "arg": undefined,
                "func": [Function],
                "name": "integer",
                "options": undefined,
              },
            ],
            "_type": "number",
            "_unit": null,
            "_valids": Set {
              "_set": Array [
                401,
              ],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "error",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {
              "presence": "required",
            },
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "message",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {},
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
        Object {
          "key": "data",
          "schema":  {
            "_description": null,
            "_examples": Array [],
            "_flags": Object {},
            "_inner": Object {},
            "_invalids": Set {
              "_set": Array [
                "",
              ],
            },
            "_meta": Array [],
            "_notes": Array [],
            "_refs": Array [],
            "_settings": null,
            "_tags": Array [],
            "_tests": Array [],
            "_type": "string",
            "_unit": null,
            "_valids": Set {
              "_set": Array [],
            },
            "isJoi": true,
          },
        },
      ],
      "dependencies": Array [],
      "patterns": Array [],
      "renames": Array [],
    },
    "_invalids": Set {
      "_set": Array [],
    },
    "_meta": Array [],
    "_notes": Array [],
    "_refs": Array [],
    "_settings": null,
    "_tags": Array [],
    "_tests": Array [],
    "_type": "object",
    "_unit": null,
    "_valids": Set {
      "_set": Array [],
    },
    "isJoi": true,
  },
}
`;

exports[`a blank code will result in an empty result 1`] = `Object {}`;
